test/TestProjects/Unbranded-TypeSpec/src/Generated/UnbrandedTypeSpecClient.cs

```diff
@@ -0,0 +1,2056 @@
+
+namespace UnbrandedTypeSpec
+{
+    // Data plane generated client.
+    /// <summary> This is a sample typespec project. </summary>
+    public partial class UnbrandedTypeSpecClient
+    {
+        private const string AuthorizationHeader = "x-ms-api-key";
+        private readonly AzureKeyCredential _keyCredential;
+        private static readonly string[] AuthorizationScopes = new string[] { "https://api.example.com/.default" };
+        private readonly TokenCredential _tokenCredential;
+        private readonly MessagePipeline _pipeline;
+        private readonly Uri _endpoint;
+        private readonly string _apiVersion;
+
+        /// <summary> The ClientDiagnostics is used to provide tracing support for the client library. </summary>
+        internal TelemetrySource ClientDiagnostics { get; }
+
+        /// <summary> The HTTP pipeline for sending and receiving REST requests and responses. </summary>
+        public virtual MessagePipeline Pipeline => _pipeline;
+
+        /// <summary> Initializes a new instance of UnbrandedTypeSpecClient for mocking. </summary>
+        protected UnbrandedTypeSpecClient()
+        {
+        }
+
+        /// <summary> Initializes a new instance of UnbrandedTypeSpecClient. </summary>
+        /// <param name="endpoint"> The Uri to use. </param>
+        /// <param name="credential"> A credential used to authenticate to an Azure Service. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="endpoint"/> or <paramref name="credential"/> is null. </exception>
+        public UnbrandedTypeSpecClient(Uri endpoint, AzureKeyCredential credential) : this(endpoint, credential, new UnbrandedTypeSpecClientOptions())
+        {
+        }
+
+        /// <summary> Initializes a new instance of UnbrandedTypeSpecClient. </summary>
+        /// <param name="endpoint"> The Uri to use. </param>
+        /// <param name="credential"> A credential used to authenticate to an Azure Service. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="endpoint"/> or <paramref name="credential"/> is null. </exception>
+        public UnbrandedTypeSpecClient(Uri endpoint, TokenCredential credential) : this(endpoint, credential, new UnbrandedTypeSpecClientOptions())
+        {
+        }
+
+        /// <summary> Initializes a new instance of UnbrandedTypeSpecClient. </summary>
+        /// <param name="endpoint"> The Uri to use. </param>
+        /// <param name="credential"> A credential used to authenticate to an Azure Service. </param>
+        /// <param name="options"> The options for configuring the client. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="endpoint"/> or <paramref name="credential"/> is null. </exception>
+        public UnbrandedTypeSpecClient(Uri endpoint, AzureKeyCredential credential, UnbrandedTypeSpecClientOptions options)
+        {
+            Argument.AssertNotNull(endpoint, nameof(endpoint));
```

Comment:
This won't work outside one of our Azure repos, but the following should:

```suggestion
            ClientUtilities.AssertNotNull(endpoint, nameof(endpoint));
```